docs/_source/guides/llms/practical_guides/fine_tune.md

```diff
@@ -1,24 +1,305 @@
+
+Underneath, you can see the happy flow for using the `ArgillaTrainer`.
+
+```python
+from argilla.feedback import ArgillaTrainer, FeedbackDataset, TrainingTask
+
+dataset = FeedbackDataset.from_huggingface(
+    repo_id="argilla/emotion"
+)
+task = TrainingTask.for_text_classification(
+    text=dataset.field_by_name("text"),
+    label=dataset.question_by_name("label"),
+)
+trainer = ArgillaTrainer(
+    dataset=dataset,
+    task=task,
+    framework="setfit"
+)
+trainer.update_config(num_iterations=1)
+trainer.train(output_dir="my_setfit_model")
+trainer.predict("This is awesome!")
+```
+
+### Supported Frameworks
+
+We plan on adding more support for other tasks and frameworks so feel free to reach out on our Slack or GitHub to help us prioritize each task.
+
+| Task/Framework                   | TRL  | OpenAI | AutoTrain | SetFit | spaCy | Transformers | PEFT |
+|:--------------------------------|:-----|:-------|:----------|:-------|:------|:-------------|:-----|
+| Text Classification             |      | ✔️      | ✔️         | ✔️      | ✔️     | ✔️            | ✔️    |
+| Supervised Fine-tuning          | ✔️    |        |           |        |       |              |      |
+| Reward Modeling                 | ✔️    |        |           |        |       |              |      |
+| Proximal Policy Optimization    | ✔️    |        |           |        |       |              |      |
+| Direct Performance Optimization | ✔️    |        |           |        |       |              |      |
+
+
+```{note}
+We also offer support for  Token Classification using our `TokenClassifcationDataset` but this is shown in [a section](/guides/train_a_model) about our older dataset-types.
+```
+
+#### Training Configs
+
+The trainer also has an `ArgillaTrainer.update_config()` method, which maps a dict with `**kwargs` to the respective framework. So, these can be derived from the underlying framework that was used to initialize the trainer. Underneath, you can find an overview of these variables for the supported frameworks.
+
+```{note}
+Note that you don't need to pass all of them directly and that the values below are their default configurations.
+```
+
+```{include} /_common/tabs/train_update_config.md
+```
+
+### The `TaskMapping`
+
+A `TaskMapping` is used to define how the data should be processed and formatted according to the associated task and framework. Each task has its own `TaskMapping.for_*`-classmethod and the data formatting can always be defined using a custom `formatting_func`. However, simpler tasks like Text Classification can also be defined using default definitions. These directly use the fields and questions from the FeedbackDataset configuration to infer how to prepare the data. Underneath you can find an overview of the `TaskMapping`` requirements.
+
+| Method                             | Content          | Return formatting_func                                     | Default           |
```

Comment:
Maybe "Default" should be e.g. "Has a default"?

Reply:
I considered that, but it slightly messes up the formatting.

Reply:
Makes sense